From bb931a18e741bb62d9252c2b494aeb7c94eb16cd Mon Sep 17 00:00:00 2001
From: Vikas Gupta <vikas.gupta@broadcom.com>
Date: Fri, 22 Nov 2013 02:19:18 -0800
Subject: [PATCH 0858/1532] netlib: fix for error counter register and added
 XLAUI support in set-frame-size API.

[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/netlogic/lib/netlib/include/netsoc_nae.h | 2 +-
 arch/mips/netlogic/lib/netlib/src/netsoc_nae.c     | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/mips/netlogic/lib/netlib/include/netsoc_nae.h b/arch/mips/netlogic/lib/netlib/include/netsoc_nae.h
index e92373c..7353455 100644
--- a/arch/mips/netlogic/lib/netlib/include/netsoc_nae.h
+++ b/arch/mips/netlogic/lib/netlib/include/netsoc_nae.h
@@ -927,7 +927,7 @@ static void __netsoc_read_mac_counters(uint64_t mac_base, dev_stat_t* stats)
 
 	stats->tx_errors = netsoc_read_mac_reg(mac_base, TX_FCS_ERROR_COUNTER) +
 				netsoc_read_mac_reg(mac_base, TX_JABBER_FRAME_COUNTER);
-	stats->rx_dropped = netsoc_read_mac_reg(mac_base, RX_DROP_PACKET_COUNTER);
+	stats->rx_dropped = netsoc_read_mac_reg(mac_base, RX_FCS_ERROR_COUNTER);
 	stats->tx_dropped = netsoc_read_mac_reg(mac_base, TX_DROP_FRAME_COUNTER);
 
 	stats->multicast = netsoc_read_mac_reg(mac_base, RX_MULTICAST_PACKET_COUNTER);
diff --git a/arch/mips/netlogic/lib/netlib/src/netsoc_nae.c b/arch/mips/netlogic/lib/netlib/src/netsoc_nae.c
index 524e476..3b68813 100644
--- a/arch/mips/netlogic/lib/netlib/src/netsoc_nae.c
+++ b/arch/mips/netlogic/lib/netlib/src/netsoc_nae.c
@@ -3089,6 +3089,9 @@ int __netsoc_config_frame_size(net_port_t *netport, uint32_t size)
 			mac_base = __netsoc_get_xgmac_base(netport);
 			netsoc_write_mac_reg(mac_base, XAUI_MAX_FRAME_LEN , ((size/4) << 16) | size);
 			break;
+		case XLAUI_IF:
+			return NETSOC_API_SUCCESS;  /*default its setting 16K*/
+			break;
 		default:
 			return -NETSOC_API_PARAM_INVALID;
 	}
-- 
1.9.1

